core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraImpl.kt

```diff
@@ -215,6 +215,23 @@ class RawInfraImpl(
         return res
     }
 
+    override fun getTrackChunkElectricalProfile(
+        trackChunk: TrackChunkId, mapping: HashMap<String, DistanceRangeMap<String>>
+    ): DistanceRangeMap<String> {
+        val chunkOffset = getTrackChunkOffset(trackChunk)
+        val trackId = getTrackFromChunk(trackChunk)
+        val trackName = getTrackSectionName(trackId)
+        assert(mapping.containsKey(trackName)) {
+            "ElectricalProfileMapping does not cover track $trackName for every PowerClass"
```

Comment:
I don't think this should be an error, the result doesn't have to cover every range

Reply:
It is, indeed, not necessary. (It has to cover the range when used to create an EnvelopeSimPath but it is already checked in it's builder and this function might be used for something else than creating an EnvelopeSimPath.)